base/broadcast.jl

```diff
@@ -279,8 +305,16 @@ ziptype(A) = typestuple(A)
 ziptype(A, B) = (Base.@_pure_meta; Iterators.Zip2{typestuple(A), typestuple(B)})
 @inline ziptype(A, B, C, D...) = Iterators.Zip{typestuple(A), ziptype(B, C, D...)}
 
-_broadcast_type(f, T::Type, As...) = Base._return_type(f, typestuple(T, As...))
-_broadcast_type(f, A, Bs...) = Base._default_eltype(Base.Generator{ziptype(A, Bs...), ftype(f, A, Bs...)})
+scalarify_nullable(x) = x
+scalarify_nullable(x::Nullable) = (x,)
```

Comment:
Actually, this makes me wonder if we could remove the `promote_op` methods for `Nullable`. Unless there's a strong reason for using them outside `Base`, the `broadcast_c(f, ::Type{Nullable}, a...)` should be enough to handle operations with `Nullable`s.

Also, if `typeof(op.(Nullable(a), Nullable(b)) == Nullable{_broadcast_type(op, a, b)}`. I guess there is no need of any of this, that is, you shouldn't need all this `scalarify_nullable` methods nor the `typestuple(a::Nullable)` I mentioned above (unless I'm missing something).